app/controllers/ems_common.rb

```diff
@@ -571,12 +571,15 @@ def recheck_authentication(id = nil)
   end
 
   def check_compliance(model)
-    emss = find_checked_items
-    if emss.empty?
-      add_flash(_("No %{model} were selected for Compliance Check") % {:model => ui_lookup(:models => model.to_s)}, :error)
+    showlist = @lastaction == "show_list"
+    ids = showlist ? find_checked_items : ([params[:id]] if model.find(params[:id]))
```

Comment:
@h-kataria right. I was afraid that this model was already deleted or that this `id` is somehow wrong and wanted to make sure we are good. If you think it to be redundant then I will change this line to:
```
ids = showlist ? find_checked_items : [params[:id]]
```